src/plugins/mer/merdeployconfiguration.cpp

```diff
@@ -178,11 +177,16 @@ void MerAddRemoveSpecialDeployStepsProjectListener::updateProject(QmakeProject *
     }
 }
 
-bool MerAddRemoveSpecialDeployStepsProjectListener::isAmbienceProject(QmakeProject *project)
+bool MerAddRemoveSpecialDeployStepsProjectListener::isAmbienceProject(Project *project)
 {
-    QmakeProFile *root = project->rootProFile();
-    return root->projectType() == ProjectType::AuxTemplate &&
-        root->variableValue(Variable::Config).contains(QLatin1String(SAILFISH_AMBIENCE_CONFIG));
+    if (qobject_cast<QmakeProject*>(project)) {
+        QmakeProFile *root = qobject_cast<QmakeProject*>(project)->rootProFile();
```

Comment:
Possibly
```
if (auto *qmakeProject = qobject_cast<QmakeProject*>(project)) {
    QmakeProFile *root = qmakeProject->rootProFile();
```

Reply:
I would not recommend assignment inside if condition. 

```
auto *qmakeProject = qobject_cast<QmakeProject*>(project);
if (qmakeProject) ...
```

should be preferred.